app/src/main/java/com/jakdor/apapp/ui/registration/RegistrationViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package com.jakdor.apapp.ui.registration
+
+import android.app.Application
+import com.jakdor.apapp.arch.BaseViewModel
+import com.jakdor.apapp.utils.RxSchedulersFacade
+import java.util.regex.Pattern
+import javax.inject.Inject
+
+class RegistrationViewModel
+@Inject constructor(application: Application,
+                    rxSchedulersFacade: RxSchedulersFacade):
+    BaseViewModel(application, rxSchedulersFacade){
+
+    fun validatePassword(password: String): String? {
```

Comment:
Powinieneś zwracać enuma ze stanem lub boola jak potrzebujesz 2 stanów

Reply:
A na widoku - Fragmencie na podstawie tego pobierać z resów przez getString(...) co ma się wyświetlić

Reply:
Jeszcze lepiej gdyby to były voidy wszystko a stany zwracane przez LiveData, to wtedy sobie to ładnie połączysz z np. z requestem do backendu ze sprawdzaniem czy login jest nie zajęty w bazie danych itd.